### DISCUSSION TITLE
Panel upload storage etc

### DISCUSSION THREAD

**Discussion by RomanKubiak** [RomanKubiak/ctrlr#110] [Ideas]
Does anyone have any good ideas on how to do that, there are limits on what github can do, and it's also not the simplest tool to use for people that want to promote their panels.

There is an option to ceate a marketplace for panels that will be accessible from the web and the app something like VCV rack has https://library.vcvrack.com/ but i don't know if that's a good idea.

Maybe there are some other options i'm unaware of, i don't want to start implementing something that will fail. The wordpress thing that is up now is a bad idea, wordpress is a nightmare to maintain.

**Comment by Ted-juh** [comment]
Are you looking for free hosting? Or is a paid alternative no problem?

**Comment by RomanKubiak** [comment]
it's more of a logistic problem, how to make this process easy enough so that anyone can do it, and that it can be moderated

**Reply by dobo365** [reply]
Exactly!!!
_I was busy to write below answer while you made this post ;-)_

**Comment by dobo365** [comment]
Yep, let's list some requirements first:
- shall we have the Panels section separate from the Ctrlr forum / main SW download or together with it?
As you say, github is fine as forum; not sure about main SW download and most probably not as Panels showcase. So we should probably say separate
- we need a way to give access to members to build their own web page for each panel. Maybe that we should simply have a predefined template for this (title, subtitle for version, main image, main text area for description, image gallery, contact info / price, contact mail / download button)
- we need a page showcasing all panels / a page showcasing deprecated panels
- other reqs (some statistic info? ...)

About solutions:
- Wordpress is used (acc to internet articles I read recently) by about 30% of the web sites and has a good SEO (search engine optimization) making the sites appearing quite early in Google searches. Maybe this is less important for us
- Wordpress seems indeed not so easy to handle. I tried making my web site on it - without much reading, almost first time of my life; hahaha - and I didn't understood how it was working to get something nice and easy. So, seems difficult to maintain.
- I tried then with Wix and I have now a complete web site for my panels. Took about 20-30 hours to build (most work is in the pictures). No need for a manual, all drag and drop. Auto alignment of objects... Nice. Seems thus OK to maintain.
- Wix contains several add-ons for file sharing, forums... but Wix seems to be poorly referenced in search engines
- In Wordpress you succeeded to provide rights to members to build their own panel page. I don't know how easy it is to achieve the same with other CMS
- You need to pay 123 EUR by year to have Wix without ads (I took the Basic fee). There are also limitations in traffic, storage and bandwidth.
- What about Wordpress?
- What about other CMS?
- ...

Can you maybe list the issues with Wordpress so we can try finding solutions? Maybe we can share the work?
Maybe that the easiest is just to stay in Wordpress (if can improve the current situation)?

**Comment by Ted-juh** [comment]
Would be nice to have the folowing requirements :

I vote for separate download page but with the option to advertise the panel in the "show and tell" category on GitHub (or another dedicated category)

- More than 1mb panel upload possible.
- Be able to re-edit the page or to upload newer versions of a panel (and keep separate versions as download.) 
- Option to send an automatically generated mail if a panel is updated. I know it's farfetched but quite handy.
- The option like on the WordPress site to send messages.
- 
There are too many CMS hosting sites like WordPress, Joomla, Drupal, Grav, Wix and so on. I'm at work so can't do a good comparison. Might do that tonight.

**Comment by RomanKubiak** [comment]
I have no problem with a web page that will present each panel, what i'd like is for the panel publishing to be simple, maybe done from within the app itself, and the webpages get generated based on the uploaded content. It's a thought

**Comment by tomlem** [comment]
RomanKubiak wrote :

> "there are limits on what github can do"

What are those limits?
Advantages usually hide disadvantages like a Janus head ...
but the reverse is also true.

I think the best solution is to keep everything together on one site (github) if possible.
It gives a larger, instant accessible, overview of all public activities associated with Ctrlr. That way, it also breeds interest in all aspects of Ctrlr.

In my opinion it's already a lot of progress the discussions have moved here.

**Comment by fbit-itc** [comment]
Dear Roman & all others - happy new year :)

On topic: it seems I always hit the perfect moment for uploading panels ... ;) ok, no New-Years panel on ctrlr.org *sniiiiffff*.

Question: maybe I'm a bit outdated, but for me github is (still) a code development and bug tracking platform. An IT tool.
Frankly: I don't see 'users' (= users of panels) willing to register to an IT platform just for publishing user-questions....

To be concrete: what's the prob with the recent wordpress solution? I mean, in details? I know WP only a very bit (running 2 little synthesi.... websites with it), in deed it's a 'nightmare' - but also is Joomla, Typo etc... and I don't see github being able to replace it in any means.
Therefore the question: where are the problematic zones: is it the 'panel-suppliers'  while creating their panel-websites or is it the general administration of WP - or both - or something different?

**Reply by dnaldoog** [reply]
this post says it all!

**Comment by trebmuh** [comment]
https://musical-artifacts.com/ has been up and running for a few years now.

**Comment by dobo365** [comment]
After some reflection and based on what I posted before and the other posts, I suggest the following:
- we use github for the Ctrlr solution downloads, the forum and its discussions, the documentation (manuals...)
- we keep using WP as today for panel upload / showcase
Advantages: we keep the development together (code, forum, discussions, downloads...) where it should be (github) and we keep the panels presentation where it is (WP) without change and thus no time to spend on this.

**Comment by RomanKubiak** [comment]
i'd like to store the panel files in something simpler then WP, and the presentation layer then could be anything. github offers github pages, i don't know how much scripting i can do in those but it's something i'll have a look into.
I'm still voting for uploading the panels from Ctrlr, i'd like to be able to sign all uploads too, for security

**Comment by Ted-juh** [comment]
AFAIK no PHP. I think JavaScript is supported though.
